internal/venice-client-common/src/main/java/com/linkedin/venice/schema/SchemaReader.java

```diff
@@ -17,5 +18,13 @@ public interface SchemaReader extends Closeable {
 
   Schema getLatestValueSchema();
 
+  /**
+   * Get the latest value schema id. This may be different from the value schema with the largest id if the superset
+   * schema is not the value schema with the largest id
+   */
   Integer getLatestValueSchemaId();
+
+  Schema getUpdateSchema(int valueSchemaId);
```

Comment:
Online producer will need the latest update schema to create the `UpdateBuilder` object using the derived schema from the superset schema

Reply:
We only need `getLatestUpdateSchema` in that, but adding just that function seemed a bit weird, hence I also added `getUpdateSchema(int valueSchemaId)` for completeness sake

Reply:
If we don't really need:
```
Integer getLatestValueSchemaId();

  Schema getUpdateSchema(int valueSchemaId);
```
I recommend to remove them since operating against schema ids are tricky and schema id related operations should mostly be left to Venice Backend components.

Reply:
`getLatestValueSchemaId` is already being used in other places. Also, I don't think it's an issue having these since `SchemaReader` is only for internal implementation. I can still remove `getUpdateSchema(int valueSchemaId)` as that is new